tests/recovery/blobmanipulator/blobmanipulator.go

```diff
@@ -274,6 +276,54 @@ func (bm *BlobManipulator) SetUpSystemUnderTest() error {
 	return nil
 }
 
+// SetUpSystemWithOneSnapshot connects or creates a kopia repo, writes random data in source directory,
+// creates snapshots of the source directory.
+func (bm *BlobManipulator) SetUpSystemWithOneSnapshot() (string, error) {
+	fileSize := 1 * 1024 * 1024
+	numFiles := 50
+	ctx := context.Background()
+
+	err := bm.writeRandomFiles(ctx, fileSize, numFiles)
+	if err != nil {
+		return "", err
+	}
+
+	// create snapshot of the data
+	bm.PathToTakeSnapshot = bm.FileHandler.GetRootDir(bm.fileWriter.DataDirectory(ctx))
```

Comment:
why not set the `bm.PathToTakeSnapshot` to the data dir ? We'd like to take the snapshot of the entire directory, right ?

```suggestion
	bm.PathToTakeSnapshot = bm.fileWriter.DataDirectory(ctx)
```

Reply:
Maybe I am missing something here. @MinyiZou I am assuming hat we'll ike to snapshot the entire directory and not just a single file.

Reply:
The structure of directories is as follows: /a/b/. In the "b" folder, there are 200 files. However, when using the bm.fileWriter.DataDirectory(ctx) method, I only receive a folder path (/a/). For this particular test scenario, it should be acceptable to capture either the entire directory as a snapshot or the 200 individual files. In case we choose to snapshot the 200 files, I will send a signal. `bm.FileHandler.GetRootDir` will return /a/b/ path if I input /a/.